### PR TITLE
fix: config support parse tls

### PR DESCRIPTION
**Pull request by TCeason**
**Summary**

* On local: if add --tls will return err:

➜  ~/database/source-code/bendsql👆 git:(main) ⚡ ./target/debug/bendsql --tls=true
path is "/home/taichong/.config/bendsql/config.toml"
not set dsn : databend://root@localhost:8000?tls=false
Welcome to BendSQL 0.18.3-fbb976e(2024-07-03T13:28:17.492814078Z).
Connecting to localhost:8000 as user root.
Error: APIError: RequestError: error sending request for url (https://localhost:8000/v1/query)

* If add it in config.toml, now it also return err:

And I think the doc in README.md is wrong. That is right.

```
➜  ~/database/source-code/bendsql👆 git:(main) ⚡ cat  /home/taichong/.config/bendsql/config.toml
[connection]
host = "127.0.0.1"

[connection.args]
tls = "true"
connect_timeout = "30"


[settings]
display_pretty_sql = true
progress_color = "green"
no_auto_complete = true
prompt = ":) "
```

➜  ~/database/source-code/bendsql👆 git:(main) ⚡ ./target/debug/bendsql                         
path is "/home/taichong/.config/bendsql/config.toml"
not set dsn : databend://root@localhost:8000?tls=true
Welcome to BendSQL 0.18.3-fbb976e(2024-07-03T13:28:17.492814078Z).
Connecting to localhost:8000 as user root.
Error: APIError: RequestError: error sending request for url (https://localhost:8000/v1/query)

Fix: #440

### CODE REVIEW

**Comment by TCeason** [comment]
cc @sundy-li

**Comment by TCeason** [comment]
I can run test  on my local. 
```sql
➜  ~/database/source-code/bendsql👆 git:(fix_440)  ./cli/test.sh http             
warning: both `/home/taichong/.cargo/config` and `/home/taichong/.cargo/config.toml` exist. Using `/home/taichong/.cargo/config`
warning: both `/home/taichong/.cargo/config` and `/home/taichong/.cargo/config.toml` exist. Using `/home/taichong/.cargo/config`
   Compiling bendsql v0.18.3 (/home/taichong/database/source-code/bendsql/cli)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 28.76s
==> Testing REST API handler
    Running test -- cli/tests/00-base.sql
23c23
< NULL	v1	2	
---
> NULL	v1	2	NULL

```
But ci failed.

```
==> Testing REST API handler
Error: APIError: RequestError: error sending request for url (https://localhost:8000/v1/query)

Stack backtrace:
   0: anyhow::error::<impl core::convert::From<E> for anyhow::Error>::from
             at /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/anyhow-1.0.86/src/backtrace.rs:27:14
   1: <core::result::Result<T,F> as core::ops::try_trait::FromResidual<core::result::Result<core::convert::Infallible,E>>>::from_residual
             at /rustc/129f3b9964af4d4a709d1383930ade12dfe7c081/library/core/src/result.rs:1964:27
   2: bendsql::session::Session::handle_query::{{closure}}
             at ./cli/src/session.rs:415:32
```

File: README.md [hunk 1]
```diff
@@ -105,8 +105,11 @@ sequentially if exists.
 ```
 ❯ cat ~/.bendsql/config.toml
 [connection]
-connect_timeout = "30s"
+host = "127.0.0.1"
 
+[connection.args]
+tls = "false"
```
Comment (everpcpc) [inline README.md:111]: why not just use 
```
tls = false
```

Comment (TCeason) [reply README.md:111]: > why not just use 
> 
> ```
> 
> tls = false
> 
> ```

args type is map<string, string>

**Review by sundy-li — APPROVED** [review]

**Review by everpcpc — APPROVED** [review]

**Review by everpcpc — CHANGES_REQUESTED** [review]
`connection.args` is intended for arguments in DSN, so it is a `map<string, string>`, tls is not included. 
Maybe it should be
```
[connection]
tls = true

[connection.args]
...
```

or
```
[connection]
...

[connection.args]
sslmode = "disable"
```

**Comment by TCeason** [comment]
> [connection]
> tls = true
> 
> [connection.args]
> ...


Good idea. Fixed. Please review the newest commit.
```
[connection]
tls = true

[connection.args]
...
```

**Review by everpcpc — APPROVED** [review]

**Pull request merged**